### PR TITLE
Sorted group list when editing contacts

### PR DESCRIPTION
**Pull request by ericnewcomer**
Inconsistent ordering of these groups was found by a RF test and this will make them deterministic.

### CODE REVIEW

**Review by norkans7 — APPROVED** [review]

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: temba/contacts/views.py [hunk 1]
```diff
@@ -438,7 +438,7 @@ def __init__(self, org, *args, **kwargs):
         )
 
         self.fields["groups"].initial = self.instance.get_groups(manual_only=True)
-        self.fields["groups"].queryset = ContactGroup.get_groups(self.org, manual_only=True)
+        self.fields["groups"].queryset = ContactGroup.get_groups(self.org, manual_only=True).order_by("name")
```
[the remark this answers is not in the document]
Comment (ericnewcomer) [reply temba/contacts/views.py:441]: You want lowercase intermingled? Do we do that elsewhere? Never been a fan and I think we've discussed this before.

[the remark this answers is not in the document]
Comment (rowanseymour) [reply temba/contacts/views.py:441]: We've generally done case-insensitive ordering.. and we also do case-insensitive uniqueness. Tho I do notice we haven't been consistent about Upper vs Lower. I personally think we should consistently ignore case everywhere.

[the remark this answers is not in the document]
Comment (ericnewcomer) [reply temba/contacts/views.py:441]: My vote is certainly for case insensitive uniqueness but not intermingling case for sorting. This of course only affects the first letter, but spring words that start with lowercase after title case is super common and feels way more right to me.

[the remark this answers is not in the document]
Comment (ericnewcomer) [reply temba/contacts/views.py:441]: Sorry, sorting words. Autocorrect. 

Case sensitive sorting is the default. I don't see why we should break that. It's far preferable to me to encourage people to follow standard lexicographic sort rules if they want them together in the list. I personally *hate* seeing lowercase words next to uppercase ones in a sorted list.

[the remark this answers is not in the document]
Comment (ericnewcomer) [reply temba/contacts/views.py:441]: I give you many wins. I'll die on this hill.

[the remark this answers is not in the document]
Comment (ericnewcomer) [reply temba/contacts/views.py:441]: Oh it's the user in me not the developer one. I will commit to opening a PR for the others today.

**Pull request merged**